subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -222,4 +222,23 @@ int zsock_getaddrinfo(const char *host, const char *service,
 	return ret;
 }
 
+#define ERR(e) case DNS_ ## e: return #e
+const char *zsock_gai_strerror(int errcode)
+{
+	switch (errcode) {
+	ERR(EAI_BADFLAGS);
+	ERR(EAI_NONAME);
+	ERR(EAI_AGAIN);
+	ERR(EAI_FAIL);
+	ERR(EAI_NODATA);
+	ERR(EAI_MEMORY);
+	ERR(EAI_SYSTEM);
+	ERR(EAI_SERVICE);
+
+	default:
+		return "unknown getaddrinfo error";
```

Comment:
Instead of this long string, and as we try to save some memory, we could just return "EAI_UNKNOWN" here.

Reply:
We try to save memory, yeah. But all error strings above match an existing error code identifier exactly. Whereas there's no such a code as EAI_UNKNOWN. So, there may be confusion. But well, that default actually isn't supposed to ever fire. So would sound good, but maybe something distinguishable still, like "EAI_???" ?

Reply:
There isn't an error code called "unknown getaddrinfo error" either, so the EAI_UNKNOWN is perfectly valid return value in this case.

Reply:
The "EAI_???" looks weird and not very clear, please use EAI_UNKNOWN here.

Reply:
> please use EAI_UNKNOWN here.

Done.